tendermint/tests/lite.rs

```diff
@@ -121,138 +120,140 @@ impl MockRequester {
 // Link to the commit that generated below JSON test files:
 // https://github.com/Shivani912/tendermint/commit/e02f8fd54a278f0192353e54b84a027c8fe31c1e
 const TEST_FILES_PATH: &str = "./tests/support/lite/";
-fn read_json_fixture(name: &str) -> String {
-    fs::read_to_string(PathBuf::from(TEST_FILES_PATH).join(name.to_owned() + ".json")).unwrap()
+fn read_json_fixture(file_path: &str) -> String {
+    fs::read_to_string(PathBuf::from(file_path)).unwrap()
 }
 
-#[test]
-fn val_set_tests_verify() {
-    let cases: TestCases =
-        serde_json::from_str(&read_json_fixture("single_step_sequential/val_set_tests")).unwrap();
-    run_test_cases(cases);
-}
+type Trusted = lite::TrustedState<SignedHeader, Header>;
 
-#[test]
-fn commit_tests_verify() {
-    let cases: TestCases =
-        serde_json::from_str(&read_json_fixture("single_step_sequential/commit_tests")).unwrap();
-    run_test_cases(cases);
+fn run_test_case(tc: &TestCase) {
+    let trusted_next_vals = tc.initial.clone().next_validator_set;
+    let mut latest_trusted =
+        Trusted::new(tc.initial.signed_header.clone().into(), trusted_next_vals);
+
+    let expects_err = match &tc.expected_output {
+        Some(eo) => eo.eq("error"),
+        None => false,
+    };
+
+    let trusting_period: std::time::Duration = tc.initial.clone().trusting_period.into();
+    let tm_now = tc.initial.now;
+    let now = tm_now.to_system_time().unwrap();
+
+    for (i, input) in tc.input.iter().enumerate() {
+        println!("i: {}, {}", i, tc.description);
+
+        let untrusted_signed_header = &input.signed_header;
+        let untrusted_vals = &input.validator_set;
+        let untrusted_next_vals = &input.next_validator_set;
+
+        match lite::verify_single(
+            latest_trusted.clone(),
+            &untrusted_signed_header.into(),
+            untrusted_vals,
+            untrusted_next_vals,
+            TrustThresholdFraction::default(),
+            trusting_period,
+            now,
+        ) {
+            Ok(new_state) => {
+                let expected_state = TrustedState::new(
+                    untrusted_signed_header.clone().into(),
+                    untrusted_next_vals.clone(),
+                );
+
+                assert_eq!(new_state, expected_state);
+                assert!(!expects_err);
+
+                latest_trusted = new_state.clone();
+            }
+            Err(_) => {
+                assert!(expects_err);
+            }
+        }
+    }
 }
+#[tokio::test]
+async fn bisection() {
+    // TODO: re-enable multi-peer tests as soon as the light client can handle these:
+    // let dir = "bisection/multi_peer";
+    // run_bisection_tests(dir).await;
```

Comment:
I think there is already an open issue about multi peer support. As the light client is currently refactored, it will probably take a little longer until we have multi peer support.